src/server/usage/classes/collector_set.js

```diff
@@ -92,6 +92,25 @@ export class CollectorSet {
     });
   }
 
+  bulkFormat(data, collectors = this._collectors) {
+    if (!Array.isArray(collectors)) {
+      throw new Error(`bulkFormat method given bad collectors parameter: ` + typeof collectors);
+    }
+
+    return data.reduce((accum, collectedData) => {
+      if (isEmpty(collectedData)) {
+        return accum;
+      }
+      const collector = collectors.find(_c => _c.type === collectedData.type);
+      const defaultFormatterForBulkUpload = result => ([
+        { type: collector.type, payload: result }
+      ]);
```

Comment:
We probably don't want to declare this function inside of a loop. I'd hoist it up and have it take a type argument.